prober/http.go

```diff
@@ -65,6 +67,58 @@ func matchRegularExpressions(reader io.Reader, httpConfig config.HTTPProbe, logg
 	return true
 }
 
+func matchCelExpressions(reader io.Reader, httpConfig config.HTTPProbe, logger log.Logger) bool {
+	body, err := io.ReadAll(reader)
+	if err != nil {
+		level.Error(logger).Log("msg", "Error reading HTTP body", "err", err)
+		return false
+	}
+
+	bodyJSON := make(map[string]interface{})
+	if err := json.Unmarshal(body, &bodyJSON); err != nil {
+		level.Error(logger).Log("msg", "Error unmarshalling HTTP body", "err", err)
+		return false
+	}
+
+	evalPayload := map[string]interface{}{
+		"body": bodyJSON,
+	}
+
+	if httpConfig.FailIfBodyJSONMatchesCel != nil {
+		result, details, err := httpConfig.FailIfBodyJSONMatchesCel.Eval(evalPayload)
```

Comment:
Question: shouldn't this be EvalContext instead? I'm not clear about the guarantees that the evaluation will terminate, and the fact that EvalContext exists suggests that there's a benefit from imposing a deadline.

Reply:
You're probably right, I've changed it to use ContextEval